hs_access_control/models/user.py

```diff
@@ -929,22 +1002,42 @@ def view_resources(self):
 
         :return: QuerySet of resource objects accessible (in any form) to user.
 
-        Held implies viewable.
+        Held implies viewable. This includes group-community-group relationships,
+        unlike GroupAccess.view_groups and GroupAccess.edit_groups.
+
+        This can be subqueried in returns, because it is lazily evaluated.
+        e.g., resource in self.uaccess.view_resources runs efficiently, because it
+        is equivalent to self.uaccess.view_resources.filter(id=resource).exists()
         """
         if not self.user.is_active:
             raise PermissionDenied("Requesting user is not active")
 
-        # need distinct due to duplicates invoked via Q expressions
-        return BaseResource.objects.filter(Q(r2urp__user=self.user) |
-                                           Q(r2grp__group__gaccess__active=True,
-                                               r2grp__group__g2ugp__user=self.user)).distinct()
+        return BaseResource.objects.filter(
+            # direct access
+            Q(r2urp__user=self.user) |
+            # access via a group
+            Q(r2grp__group__gaccess__active=True,
+              r2grp__group__g2ugp__user=self.user) |
+            # access via an unprivileged peer group in a community
+            Q(r2grp__group__gaccess__active=True,
```

Comment:
This filter condition `r2grp__group__gaccess__active=True` probably not needed if we are looking for groups that are part of a community.

Reply:
Same comment as before. We need to handle all possible user behaviors.